libhoney.go

```diff
@@ -169,12 +169,16 @@ func (c *Config) getDataset() string {
 }
 
 func (c *Config) IsClassic() bool {
-	if len(c.APIKey) == 0 {
+	return IsClassicKey(c.APIKey)
+}
+
+func IsClassicKey(key string) bool {
```

Comment:
```suggestion
func IsClassicAPIKey(key string) bool {
```

nit: i have a slight preference for calling it `IsClassicAPIKey` to match `VerifyAPIKey` in this same file, but I think this looks great!

Reply:
beeline-go already has a `IsClassicKey` function which is why I named it this 🤷🏻 

Naming is hard 🙃

Reply:
Got it, sounds good then!